pkg/provider/utils.go

```diff
@@ -35,3 +38,12 @@ func unsupportedWarning(message string) diag.Diagnostics {
 		},
 	}
 }
+
+func fromErr(err error, summary string, args ...any) diag2.Diagnostics {
```

Comment:
This function is quite dangerous. 
When we use `fromErr`, it will remove all the other Diagnostics that were created during any Method (can be warnings etc.) and will return a clean New Error Diagnostic that returns a given error. 

Please use any [Diagnostics](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/diag#Diagnostics) method to add the error (AddError ideally) instead of completely overriding the errors.

Reply:
Hi Serdar, could you please provide more information about the statement "This function is quite dangerous"? As I understand it, this function returns "Diagnostics," which will be appended to `resp.Diagnostics` using the `Append` method. It should not overwrite any existing diagnostics. The purpose of `fromErr` is to convert errors into `Diagnostics` by accepting an error type parameter. On the other hand, `Diagnostics.AddError` does not accept an error parameter, which means you need to convert the error to a string in your subsequent code